packages/common/src/store/purchase-content/sagas.ts

```diff
@@ -300,6 +300,28 @@ function* pollForPurchaseConfirmation({
     currentUserId,
     isSourceTrack: true
   })
+
+  if (contentType === PurchaseableContentType.ALBUM) {
+    const { metadata } = yield* call(getContentInfo, {
+      contentId,
+      contentType
+    })
+    if (
+      'playlist_contents' in metadata &&
+      metadata.playlist_contents.track_ids
+    ) {
+      yield* all(
+        Array.from(metadata.playlist_contents.track_ids).map((trackId) => {
+          return call(pollGatedContent, {
```

Comment:
Wondering if there's any bad race here where the playlist updates while it's being purchased?
I don't think it's going to be very common though

Reply:
i think we would've seen that manifest with track purchases. pretty sure we don't fetch the content in this flow until here.

Reply:
or wait, what race condition were you imagining?

Reply:
again, likely rare, but I was thinking:

1. I go to album page
2. Artist removes track B from the page
3. I purchase the album
4. This loop has the stale track list and waits for me to get access to B but I don't get it because it wasn't in the album at the time I actually made my purchase

Reply:
ohhhhh....
i'm not worried about the part where the user might not get access - that's just unlucky.
but i am worried that the user would be stuck infinitely waiting for access on a track they will never get.
i think if we force-fetch the album when the user opens the purchase modal, that will probably be good enough?

Reply:
Yeah I think that'd do it. If you want to be extra sure, fetch it again right before this check so you always check the updated list for the polling check

Reply:
this is resolved now that we're fetching and not polling! woo!